### ISSUE TITLE
[BUG] Step ID 'changed-files' Is Out of Date

### ISSUE THREAD

**Issue by patgmiller** [tj-actions/eslint-changed-files#1900]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does this issue exist in the latest version?

- [X] I'm using the latest release

### Describe the bug?

The latest version of `eslint-changed-files` doesn't use the most recent version of `changed-files` and it is failing for a local `.git` repo. Doesn't appear to matter what options you use. 

**Job Definition**

```yaml
  eslint-src:
    runs-on: ubuntu-latest
    container:
      image: ${{ inputs.image }}
    environment: ${{ inputs.environment }}
    steps:
      - uses: actions/checkout@v4
        with:
          fetch-depth: 0

      - name: Restore Node Cache (Hosting)
        uses: actions/cache/restore@v4
        with:
          path: |
            node_modules/
          key: ${{ runner.os }}-build-host-${{ hashFiles('package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-build-host-
          fail-on-cache-miss: true

      - name: list files
        run: |
          env
          pwd
          ls -la .

      - name: Run eslint on changed files
        uses: tj-actions/eslint-changed-files@v25.1.0
        with:
          # path: src
          ignore_patterns: functions/**
          config_path: .eslintrc.js
          extra_args: --max-warnings=0
          reporter: github-check
```

fails with following error 

```
  Error: Unable to locate the git repository in the given path: /__w/fleet-adventures/fleet-adventures.
   Please run actions/checkout before this action (Make sure the 'path' input is correct).
   If you intend to use Github's REST API note that only pull_request* events are supported. Current event is "push".
```

however as the job steps shows in the previous step it displays the `env`, `pwd`, and lists dir contents `ls -la .` as the following

```shell
GITHUB_STATE=/__w/_temp/_runner_file_commands/save_state_419f50c0-e6f0-4891-bdfd-347faf1c7ff5
CI=true
NODE_VERSION=18.20.3
HOSTNAME=47fa0114d4e3
YARN_VERSION=1.22.19
GITHUB_ENV=/__w/_temp/_runner_file_commands/set_env_419f50c0-e6f0-4891-bdfd-347faf1c7ff5
RUNNER_ENVIRONMENT=github-hosted
HOME=/github/home
GITHUB_EVENT_PATH=/github/workflow/event.json
RUNNER_TEMP=/__w/_temp
GITHUB_REPOSITORY_OWNER=Fleetmaid
GITHUB_RETENTION_DAYS=1
GITHUB_REPOSITORY_OWNER_ID=[16](https://github.com/Fleetmaid/fleet-adventures/actions/runs/9492513071/job/26159833336#step:4:17)0766473
GITHUB_HEAD_REF=
GITHUB_GRAPHQL_URL=https://api.github.com/graphql
GITHUB_API_URL=https://api.github.com
RUNNER_OS=Linux
GITHUB_WORKFLOW=.github/workflows/_main.yaml
RUNNER_DEBUG=1
GITHUB_RUN_ID=9492513071
GITHUB_BASE_REF=
GITHUB_REF_TYPE=branch
GITHUB_WORKFLOW_SHA=2f9131324fbf96246e864ebad7db4181ff40ed13
GITHUB_WORKFLOW_REF=Fleetmaid/fleet-adventures/.github/workflows/_main.yaml@refs/heads/1450-feature--start-enforcing-eslint
GITHUB_ACTION_REPOSITORY=
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
RUNNER_TOOL_CACHE=/__t
GITHUB_RUN_NUMBER=1047
GITHUB_TRIGGERING_ACTOR=patgmiller
GITHUB_ACTION=__run
RUNNER_ARCH=X64
GITHUB_REPOSITORY=Fleetmaid/fleet-adventures
GITHUB_REF_NAME=1450-feature--start-enforcing-eslint
RUNNER_NAME=GitHub Actions 8
GITHUB_ACTION_REF=
GITHUB_ACTIONS=true
GITHUB_REPOSITORY_ID=445251522
GITHUB_REF_PROTECTED=false
GITHUB_JOB=eslint-src
GITHUB_WORKSPACE=/__w/fleet-adventures/fleet-adventures
GITHUB_SHA=2f9131324fbf96246e864ebad7db4181ff40ed13
GITHUB_RUN_ATTEMPT=1
GITHUB_REF=refs/heads/1450-feature--start-enforcing-eslint
GITHUB_ACTOR=patgmiller
GITHUB_ACTOR_ID=3914379
GITHUB_PATH=/__w/_temp/_runner_file_commands/add_path_419f50c0-e6f0-4891-bdfd-347faf1c7ff5
RUNNER_WORKSPACE=/__w/fleet-adventures
***
GITHUB_EVENT_NAME=push
GITHUB_SERVER_URL=https://github.com
GITHUB_OUTPUT=/__w/_temp/_runner_file_commands/set_output_419f50c0-e6f0-4891-bdfd-347faf1c7ff5
REFRESHED_AT=$(date +%Y-%m-%d)
GITHUB_STEP_SUMMARY=/__w/_temp/_runner_file_commands/step_summary_419f50c0-e6f0-4891-bdfd-347faf1c7ff5
/__w/fleet-adventures/fleet-adventures
total 892
drwxr-xr-x 11 1001  127   4096 Jun 13 02:08 .
drwxr-xr-x  3 1001  127   4096 Jun 13 02:08 ..
-rw-r--r--  1 root root    274 Jun 13 02:08 .editorconfig
-rw-r--r--  1 root root    626 Jun 13 02:08 .eslintrc.js
drwxr-xr-x  2 root root   4096 Jun 13 02:08 .firebase
-rw-r--r--  1 root root    443 Jun 13 02:08 .firebaserc
drwxr-xr-x  2 root root   4096 Jun 13 02:08 .fleet
drwxr-xr-x  8 root root   4096 Jun 13 02:08 .git
-rw-r--r--  1 root root     66 Jun 13 02:08 .gitattributes
drwxr-xr-x  4 root root   4096 Jun 13 02:08 .github
-rw-r--r--  1 root root    943 Jun 13 02:08 .gitignore
-rw-r--r--  1 root root      9 Jun 13 02:08 .nvmrc
drwxr-xr-x  2 root root   4096 Jun 13 02:08 .vscode
-rw-r--r--  1 root root   1061 Jun 13 02:08 README.md
-rw-r--r--  1 root root   9002 Jun 13 02:08 angular.json
-rw-r--r--  1 root root   33[17](https://github.com/Fleetmaid/fleet-adventures/actions/runs/9492513071/job/26159833336#step:4:18) Jun 13 02:08 firebase.json
-rw-r--r--  1 root root  33203 Jun 13 02:08 firestore.indexes.json
-rw-r--r--  1 root root   3913 Jun 13 02:08 firestore.rules
drwxr-xr-x  3 root root   4096 Jun 13 02:08 functions
-rw-r--r--  1 root root   1594 Jun 13 02:08 karma-cli.conf.js
-rw-r--r--  1 root root   1433 Jun 13 02:08 karma.conf.js
drwxr-xr-x  3 root root   4096 Jun 13 02:08 node_patches
-rw-r--r--  1 root root 745394 Jun 13 02:08 package-lock.json
-rw-r--r--  1 root root   4670 Jun 13 02:08 package.json
-rw-r--r--  1 root root    172 Jun 13 02:08 proxy.conf.json
drwxr-xr-x  2 root root   4096 Jun 13 02:08 scripts
drwxr-xr-x  6 root root   4096 Jun 13 02:08 src
-rw-r--r--  1 root root    175 Jun 13 02:08 storage.rules
-rw-r--r--  1 root root    [28](https://github.com/Fleetmaid/fleet-adventures/actions/runs/9492513071/job/26159833336#step:4:29)7 Jun 13 02:08 tsconfig.app.json
-rw-r--r--  1 root root    955 Jun 13 02:08 tsconfig.json
-rw-r--r--  1 root root    347 Jun 13 02:08 tsconfig.spec.json
``` 

The `.git` directory is there 

### To Reproduce

Only need to run the latest version of the action

### What OS are you seeing the problem on?

ubuntu-latest or ubuntu-22.04

### Expected behavior?

It should find the local `.git` folder

I was able to test the latest version of `changed-files` locally by running with the following environment setup. 

**File:** `.env` in root of `changed-files`
```
GITHUB_WORKSPACE="/Users/patrick/src/fleetmaid.com/fleet-adventures"
GITHUB_ACTION_PATH="/Users/patrick/src/fleetmaid.com/fleet-adventures"
INPUT_USE_REST_API=false
INPUT_API_URL=''
INPUT_DIFF_RELATIVE=true
INPUT_DIR_NAMES=false
INPUT_DIR_NAMES_DELETED_FILES_INCLUDE_ONLY_DELETED_DIRS=false
INPUT_DIR_NAMES_EXCLUDE_CURRENT_DIR=false
INPUT_DIR_NAMES_INCLUDE_FILES=''
INPUT_DIR_NAMES_INCLUDE_FILES_SEPARATOR="\n"
INPUT_DIR_NAMES_MAX_DEPTH=''
INPUT_INCLUDE_ALL_OLD_NEW_RENAMED_FILES=false
INPUT_ESCAPE_JSON=true
INPUT_EXCLUDE_SUBMODULES=false
INPUT_FAIL_ON_INITIAL_DIFF_ERROR=false
INPUT_FAIL_ON_SUBMODULE_DIFF_ERROR=false
INPUT_FETCH_ADDITIONAL_SUBMODULE_HISTORY=false
INPUT_FETCH_DEPTH=25
INPUT_FETCH_MISSING_HISTORY_MAX_RETRIES=
INPUT_MATRIX=false
INPUT_SAFE_OUTPUT=true
INPUT_SINCE_LAST_REMOTE_COMMIT=false
INPUT_WRITE_OUTPUT_FILES=false
INPUT_OUTPUT_RENAMED_FILES_AS_DELETED_AND_ADDED=false
INPUT_RECOVER_DELETED_FILES=false
INPUT_SKIP_INITIAL_FETCH=true
INPUT_NEGATION_PATTERNS_FIRST=false
INPUT_USE_POSIX_PATH_SEPARATOR=false
INPUT_JSON=false
INPUT_OLD_NEW_FILES_SEPARATOR=,
INPUT_OLD_NEW_SEPARATOR="\n"
INPUT_QUOTEPATH=true
INPUT_SEPARATOR=,
INPUT_BASE_SHA=
```

**Run:** using node v20
```shell
node --env-file .env lib/main.js
```

**Local test log**
```shell
node --env-file .env lib/main.js
::group::changed-files
::debug::Env: {%0A  "TERM_SESSION_ID": "w0t2p0:8CB58228-C6CF-4305-BDAC-AF0B0DC8FE80",%0A  "SSH_AUTH_SOCK": "/private/tmp/com.apple.launchd.aUMQawhBCV/Listeners",%0A  "LC_TERMINAL_VERSION": "3.5.1",%0A  "COLORFGBG": "7;0",%0A  "ITERM_PROFILE": "Default",%0A  "SQLITE_EXEMPT_PATH_FROM_VNODE_GUARDS": "/Users/patrick/Library/WebKit/Databases",%0A  "XPC_FLAGS": "0x0",%0A  "LANG": "en_US.UTF-8",%0A  "PWD": "/Users/patrick/src/github.com/tj-actions/changed-files",%0A  "SHELL": "/bin/zsh",%0A  "__CFBundleIdentifier": "com.googlecode.iterm2",%0A  "TERM_FEATURES": "T3LrMSc7UUw9Ts3BFGsSyHNoSxF",%0A  "TERM_PROGRAM_VERSION": "3.5.1",%0A  "TERM_PROGRAM": "iTerm.app",%0A  "PATH": "/Users/patrick/bin/google-cloud-sdk/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/patrick/.nvm/versions/node/v20.14.0/bin:/Users/patrick/bin:/Users/patrick/go/bin:/usr/local/opt/gnu-getopt/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/local/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/bin:/var/run/com.apple.security.cryptexd/codex.system/bootstrap/usr/appleinternal/bin:/usr/local/go/bin:/Applications/iTerm.app/Contents/Resources/utilities",%0A  "LC_TERMINAL": "iTerm2",%0A  "COLORTERM": "truecolor",%0A  "COMMAND_MODE": "unix2003",%0A  "TERM": "xterm-256color",%0A  "TERMINFO_DIRS": "/Applications/iTerm.app/Contents/Resources/terminfo:/usr/share/terminfo",%0A  "HOME": "/Users/patrick",%0A  "USER": "patrick",%0A  "XPC_SERVICE_NAME": "0",%0A  "LOGNAME": "patrick",%0A  "__CF_USER_TEXT_ENCODING": "0x1F5:0x0:0x0",%0A  "ITERM_SESSION_ID": "w0t2p0:8CB58228-C6CF-4305-BDAC-AF0B0DC8FE80",%0A  "SHLVL": "1",%0A  "OLDPWD": "/Users/patrick/src/github.com/tj-actions/eslint-changed-files",%0A  "GOPATH": "/Users/patrick/go",%0A  "ZSH": "/Users/patrick/.oh-my-zsh",%0A  "NVM_DIR": "/Users/patrick/.nvm",%0A  "NVM_CD_FLAGS": "-q",%0A  "NVM_BIN": "/Users/patrick/.nvm/versions/node/v20.14.0/bin",%0A  "NVM_INC": "/Users/patrick/.nvm/versions/node/v20.14.0/include/node",%0A  "PAGER": "less",%0A  "LESS": "-R",%0A  "LSCOLORS": "Gxfxcxdxbxegedabagacad",%0A  "LS_COLORS": "di=1;36:ln=35:so=32:pi=33:ex=31:bd=34;46:cd=34;43:su=30;41:sg=30;46:tw=30;42:ow=30;43",%0A  "HOMEBREW_PREFIX": "/opt/homebrew",%0A  "HOMEBREW_CELLAR": "/opt/homebrew/Cellar",%0A  "HOMEBREW_REPOSITORY": "/opt/homebrew",%0A  "MANPATH": "/Users/patrick/.nvm/versions/node/v20.14.0/share/man:/opt/homebrew/share/man::",%0A  "INFOPATH": "/opt/homebrew/share/info:",%0A  "SSH_KEY_PATH": "~/.ssh/id_ed25519",%0A  "CLOUDSDK_CONTAINER_USE_V1_API_CLIENT": "false",%0A  "EDITOR": "/usr/bin/vim",%0A  "KUBE_EDITOR": "/usr/bin/vim",%0A  "KUBECONFIG": "",%0A  "HELM_HOME": "",%0A  "separator": ",",%0A  "TMPDIR": "/var/folders/yp/j3wp4jsx6qb10q7l8b3hh7lc0000gn/T/",%0A  "_": "/Users/patrick/.nvm/versions/node/v20.14.0/bin/node",%0A  "GITHUB_ACTION_PATH": "/Users/patrick/src/fleetmaid.com/fleet-adventures",%0A  "GITHUB_WORKSPACE": "/Users/patrick/src/fleetmaid.com/fleet-adventures",%0A  "INPUT_": "",%0A  "INPUT_API_URL": "",%0A  "INPUT_BASE_SHA": "",%0A  "INPUT_DIFF_RELATIVE": "true",%0A  "INPUT_DIR_NAMES": "false",%0A  "INPUT_DIR_NAMES_DELETED_FILES_INCLUDE_ONLY_DELETED_DIRS": "false",%0A  "INPUT_DIR_NAMES_EXCLUDE_CURRENT_DIR": "false",%0A  "INPUT_DIR_NAMES_INCLUDE_FILES": "",%0A  "INPUT_DIR_NAMES_INCLUDE_FILES_SEPARATOR": "\n",%0A  "INPUT_DIR_NAMES_MAX_DEPTH": "",%0A  "INPUT_ESCAPE_JSON": "true",%0A  "INPUT_EXCLUDE_SUBMODULES": "false",%0A  "INPUT_FAIL_ON_INITIAL_DIFF_ERROR": "false",%0A  "INPUT_FAIL_ON_SUBMODULE_DIFF_ERROR": "false",%0A  "INPUT_FETCH_ADDITIONAL_SUBMODULE_HISTORY": "false",%0A  "INPUT_FETCH_DEPTH": "25",%0A  "INPUT_FETCH_MISSING_HISTORY_MAX_RETRIES": "",%0A  "INPUT_INCLUDE_ALL_OLD_NEW_RENAMED_FILES": "false",%0A  "INPUT_JSON": "false",%0A  "INPUT_MATRIX": "false",%0A  "INPUT_NEGATION_PATTERNS_FIRST": "false",%0A  "INPUT_OLD_NEW_FILES_SEPARATOR": ",",%0A  "INPUT_OLD_NEW_SEPARATOR": "\n",%0A  "INPUT_OUTPUT_RENAMED_FILES_AS_DELETED_AND_ADDED": "false",%0A  "INPUT_QUOTEPATH": "true",%0A  "INPUT_RECOVER_DELETED_FILES": "false",%0A  "INPUT_SAFE_OUTPUT": "true",%0A  "INPUT_SEPARATOR": ",",%0A  "INPUT_SINCE_LAST_REMOTE_COMMIT": "false",%0A  "INPUT_SKIP_INITIAL_FETCH": "true",%0A  "INPUT_USE_POSIX_PATH_SEPARATOR": "false",%0A  "INPUT_USE_REST_API": "false",%0A  "INPUT_WRITE_OUTPUT_FILES": "false"%0A}
::debug::Env: {%0A  "GITHUB_REF_NAME": "",%0A  "GITHUB_REF": "",%0A  "GITHUB_WORKSPACE": "/Users/patrick/src/fleetmaid.com/fleet-adventures"%0A}
::debug::Inputs: {%0A  "files": "",%0A  "filesSeparator": "",%0A  "filesFromSourceFile": "",%0A  "filesFromSourceFileSeparator": "",%0A  "filesYaml": "",%0A  "filesYamlFromSourceFile": "",%0A  "filesYamlFromSourceFileSeparator": "",%0A  "filesIgnore": "",%0A  "filesIgnoreSeparator": "",%0A  "filesIgnoreFromSourceFile": "",%0A  "filesIgnoreFromSourceFileSeparator": "",%0A  "filesIgnoreYaml": "",%0A  "filesIgnoreYamlFromSourceFile": "",%0A  "filesIgnoreYamlFromSourceFileSeparator": "",%0A  "failOnInitialDiffError": false,%0A  "failOnSubmoduleDiffError": false,%0A  "separator": ",",%0A  "sha": "",%0A  "baseSha": "",%0A  "since": "",%0A  "until": "",%0A  "path": "",%0A  "quotepath": true,%0A  "diffRelative": true,%0A  "sinceLastRemoteCommit": false,%0A  "recoverDeletedFiles": false,%0A  "recoverDeletedFilesToDestination": "",%0A  "recoverFiles": "",%0A  "recoverFilesSeparator": "",%0A  "recoverFilesIgnore": "",%0A  "recoverFilesIgnoreSeparator": "",%0A  "includeAllOldNewRenamedFiles": false,%0A  "oldNewSeparator": "\n",%0A  "oldNewFilesSeparator": ",",%0A  "skipInitialFetch": true,%0A  "fetchAdditionalSubmoduleHistory": false,%0A  "dirNamesDeletedFilesIncludeOnlyDeletedDirs": false,%0A  "excludeSubmodules": false,%0A  "usePosixPathSeparator": false,%0A  "tagsPattern": "",%0A  "tagsIgnorePattern": "",%0A  "dirNames": false,%0A  "dirNamesExcludeCurrentDir": false,%0A  "dirNamesIncludeFiles": "",%0A  "dirNamesIncludeFilesSeparator": "\n",%0A  "json": false,%0A  "escapeJson": true,%0A  "safeOutput": true,%0A  "writeOutputFiles": false,%0A  "outputDir": "",%0A  "outputRenamedFilesAsDeletedAndAdded": false,%0A  "token": "",%0A  "apiUrl": "",%0A  "negationPatternsFirst": false,%0A  "useRestApi": false,%0A  "fetchDepth": 25%0A}
::debug::Github Context: {%0A  "payload": {},%0A  "runNumber": null,%0A  "runId": null,%0A  "apiUrl": "https://api.github.com",%0A  "serverUrl": "https://github.com",%0A  "graphqlUrl": "https://api.github.com/graphql"%0A}
::debug::Working directory: /Users/patrick/src/fleetmaid.com/fleet-adventures
::debug::Has git directory: true
::debug::Input file patterns: %0A
::debug::File patterns:
::debug::Yaml file patterns: {}
Using local .git directory
Running on a push event...
::debug::Getting current SHA...
::debug::Current SHA: 2f9131324fbf96246e864ebad7db4181ff40ed13
::debug::Getting previous SHA...
::debug::Getting previous SHA for last remote commit...
::debug::Previous SHA: 3144a21816429d8dc687d581408ede04936403fc
::debug::Target branch: 1450-feature--start-enforcing-eslint
::debug::Current branch: 1450-feature--start-enforcing-eslint
Retrieving changes between 3144a21816429d8dc687d581408ede04936403fc (1450-feature--start-enforcing-eslint) → 2f9131324fbf96246e864ebad7db4181ff40ed13 (1450-feature--start-enforcing-eslint)
::debug::All diff files: {"A":[],"C":[],"D":[],"M":[".github/workflows/share-test.yml"],"R":[],"T":[],"U":[],"X":[]}
All Done!
::endgroup::
::group::changed-files-all
::debug::Dir names include file patterns: []
::debug::Added files: {"paths":"","count":"0"}

::set-output name=added_files::

::set-output name=added_files_count::0
::debug::Dir names include file patterns: []
::debug::Copied files: {"paths":"","count":"0"}

::set-output name=copied_files::

::set-output name=copied_files_count::0
::debug::Dir names include file patterns: []
::debug::Modified files: {"paths":".github/workflows/share-test.yml","count":"1"}

::set-output name=modified_files::.github/workflows/share-test.yml

::set-output name=modified_files_count::1
::debug::Dir names include file patterns: []
::debug::Renamed files: {"paths":"","count":"0"}

::set-output name=renamed_files::

::set-output name=renamed_files_count::0
::debug::Dir names include file patterns: []
::debug::Type changed files: {"paths":"","count":"0"}

::set-output name=type_changed_files::

::set-output name=type_changed_files_count::0
::debug::Dir names include file patterns: []
::debug::Unmerged files: {"paths":"","count":"0"}

::set-output name=unmerged_files::

::set-output name=unmerged_files_count::0
::debug::Dir names include file patterns: []
::debug::Unknown files: {"paths":"","count":"0"}

::set-output name=unknown_files::

::set-output name=unknown_files_count::0
::debug::Dir names include file patterns: []
::debug::All changed and modified files: {"paths":".github/workflows/share-test.yml","count":"1"}

::set-output name=all_changed_and_modified_files::.github/workflows/share-test.yml

::set-output name=all_changed_and_modified_files_count::1
::debug::Dir names include file patterns: []
::debug::All changed files: {"paths":".github/workflows/share-test.yml","count":"1"}

::set-output name=all_changed_files::.github/workflows/share-test.yml

::set-output name=all_changed_files_count::1

::set-output name=any_changed::true
::debug::Dir names include file patterns: []
::debug::All other changed files: {"paths":".github/workflows/share-test.yml","count":"1"}

::set-output name=only_changed::false
::debug::other_changed_files: []

::set-output name=other_changed_files::

::set-output name=other_changed_files_count::0
::debug::Dir names include file patterns: []
::debug::All modified files: {"paths":".github/workflows/share-test.yml","count":"1"}

::set-output name=all_modified_files::.github/workflows/share-test.yml

::set-output name=all_modified_files_count::1

::set-output name=any_modified::true
::debug::Dir names include file patterns: []

::set-output name=only_modified::false
::debug::other_modified_files: []

::set-output name=other_modified_files::

::set-output name=other_modified_files_count::0
::debug::Dir names include file patterns: []
::debug::Deleted files: {"paths":"","count":"0"}

::set-output name=deleted_files::

::set-output name=deleted_files_count::0

::set-output name=any_deleted::false
::debug::Dir names include file patterns: []

::set-output name=only_deleted::false
::debug::other_deleted_files: []

::set-output name=other_deleted_files::

::set-output name=other_deleted_files_count::0
All Done!
::endgroup::
```

### Relevant log output

[job-logs.txt](https://github.com/user-attachments/files/15813665/job-logs.txt)

### Has all relevant logs been included?

- [X] I've included all relevant logs

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by patgmiller** [comment]
Actually, it's not that the `changed-files` is out of date, it's something with the git config when `changed-files` actually runs. 

```shell
  /usr/bin/git rev-parse --is-inside-work-tree
  fatal: detected dubious ownership in repository at '/__w/fleet-adventures/fleet-adventures'
  To add an exception for this directory, call:
  	git config --global --add safe.directory /__w/fleet-adventures/fleet-adventures
  ##[debug]Has git directory: false
  ##[debug]files patterns: **/*.{ts,tsx,js,jsx}
  ##[debug]files ignore patterns: !functions/**
  ##[debug]Input file patterns: 
  ##[debug]**/*.{ts,tsx,js,jsx}
  ##[debug]!functions/**
  ##[debug]File patterns: **/*.{ts,tsx,js,jsx},!functions/**
  ##[debug]Yaml file patterns: {}
  Error: Unable to locate the git repository in the given path: /__w/fleet-adventures/fleet-adventures/src.
   Please run actions/checkout before this action (Make sure the 'path' input is correct).
   If you intend to use Github's REST API note that only pull_request* events are supported. Current event is "push".
```

The exec command to detect the `.git` dir fails even though the original checkout sets `safe` 

```shell
Run actions/checkout@v4
  with:
    fetch-depth: 0
    repository: Fleetmaid/fleet-adventures
    token: ***
    ssh-strict: true
    ssh-user: git
    persist-credentials: true
    clean: true
    sparse-checkout-cone-mode: true
    fetch-tags: false
    show-progress: true
    lfs: false
    submodules: false
    set-safe-directory: true
```

**Comment by patgmiller** [comment]
so @jackton1 the issue is definitely related to file / dir permissions when running the `.exec` `git` command. I recently changed the image which tests are running in and that image runs with the root user after installing `chrome`. I made the switch so we can run `ng test` in the build pipeline. I was able to do a work around and run `chown` to resolve it, but I don't love it and I will likely build my own image or find another image later without the root. 

It's up to you on if you decide to follow suite with the `github` `actions/checkout` and include, or just do it, `set-safe-directory` by default, since some images may require running with diff users or root. But this one is clearly an edge case.
